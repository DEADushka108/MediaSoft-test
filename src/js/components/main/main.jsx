import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Header from '../header/header';
import Footer from '../footer/footer';

const Main = () => {
  return <Fragment>
    <Header/>
    <main className="page-main">
      <h1 className="visually-hidden">Главная страница тестового задания</h1>
      
    </main>

    <Footer/>

  </Fragment>
}

export default Main;
