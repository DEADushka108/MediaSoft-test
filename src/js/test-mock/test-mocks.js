export * from './const/const';
export * from './ships/ships';
export * from './store/store';
