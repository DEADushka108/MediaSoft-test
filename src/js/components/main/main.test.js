import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {Provider} from 'react-redux';
import {shipsList, store, MAX_CARD_COUNT} from '../../test-mock/test-mocks';
import {Main} from './main';

it(`Should Main component render correctly`, () => {
  const tree = renderer
    .create((
      <Provider store={store}>
        <MemoryRouter>
          <Main
            filteredShipsList={shipsList}
            cardCount={MAX_CARD_COUNT}
          />
        </MemoryRouter>
      </Provider>
    ), {
      createNodeMock() {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
