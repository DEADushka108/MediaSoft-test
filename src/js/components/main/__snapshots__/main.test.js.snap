// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Main component render correctly 1`] = `
Array [
  <header
    className="page-header"
  >
    <div
      className="page-header__wrapper"
    >
      <a
        className="page-header__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <nav
        className="main-nav"
      >
        <ul
          className="main-nav__list"
        >
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link main-nav__link--active"
              href="/"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#home-page"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Shop
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/contact"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#social-group"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Contact
              </span>
            </a>
          </li>
          <li
            className="main-nav__item"
          >
            <a
              className="main-nav__link"
              href="/cart"
              onClick={[Function]}
            >
              <svg
                className="main-nav__icon"
              >
                <use
                  xlinkHref="#shopping-cart"
                />
              </svg>
              <span
                className="main-nav__text"
              >
                Cart
              </span>
              <span
                className="main-nav__sub"
              >
                2
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </header>,
  <main
    className="page-main"
  >
    <div
      className="page-main__wrapper"
    >
      <h1
        className="visually-hidden"
      >
        Главная страница тестового задания
      </h1>
      <div
        className="page-content"
      >
        <section
          className="catalog"
        >
          <h2
            className="catalog__title visually-hidden"
          >
            Catalog
          </h2>
          <div
            className="catalog__wrapper"
          >
            <ul
              className="catalog__race-list"
            >
              <li
                className="catalog__race-item catalog__race-item--active"
                onClick={[Function]}
              >
                <a
                  className="catalog__race-link"
                >
                  All
                </a>
              </li>
              <li
                className="catalog__race-item "
                onClick={[Function]}
              >
                <a
                  className="catalog__race-link"
                >
                  Alien
                </a>
              </li>
              <li
                className="catalog__race-item "
                onClick={[Function]}
              >
                <a
                  className="catalog__race-link"
                >
                  Human
                </a>
              </li>
              <li
                className="catalog__race-item "
                onClick={[Function]}
              >
                <a
                  className="catalog__race-link"
                >
                  Mixed
                </a>
              </li>
            </ul>
            <ul
              className="catalog__sort-list"
            >
              <li
                className="catalog__sort-item "
                onClick={[Function]}
              >
                <a
                  className="catalog__sort-link"
                >
                  <svg
                    className="catalog__sort-icon"
                  >
                    <use
                      xlinkHref="#sort-up"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    Ascending
                  </span>
                </a>
              </li>
              <li
                className="catalog__sort-item "
                onClick={[Function]}
              >
                <a
                  className="catalog__sort-link"
                >
                  <svg
                    className="catalog__sort-icon"
                  >
                    <use
                      xlinkHref="#sort-down"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    Descending
                  </span>
                </a>
              </li>
              <li
                className="catalog__sort-item"
                onClick={[Function]}
              >
                <a
                  className="catalog__sort-link"
                >
                  <svg
                    className="catalog__sort-icon"
                  >
                    <use
                      xlinkHref="#refresh"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    Refresh
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            className="catalog__ships-list"
          >
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/battlestar-galactica-preview.png"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/3"
                    onClick={[Function]}
                  >
                    Battlestar Galactica
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  234122
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                In cart
                 
              </button>
            </article>
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/borgs-cube-preview.jpg"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/4"
                    onClick={[Function]}
                  >
                    Borg's Cube
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  1002000
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                In cart
                 
              </button>
            </article>
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/death-star-preview.png"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/5"
                    onClick={[Function]}
                  >
                    Death Star
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  1000000
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                Add
                 
              </button>
            </article>
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/eagle-5-preview.jpg"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/6"
                    onClick={[Function]}
                  >
                    Eagle-5
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  100
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                Add
                 
              </button>
            </article>
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/flying-dishes-preview.jpg"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/7"
                    onClick={[Function]}
                  >
                    Flying Dish
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  3400000
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                Add
                 
              </button>
            </article>
            <article
              className="small-card"
            >
              <div
                className="small-card__image"
                onClick={[Function]}
              >
                <img
                  className="small-card__img"
                  src="./img/content/golden-heart-preview.png"
                />
              </div>
              <div
                className="small-card__wrapper"
              >
                <h3
                  className="small-card__name"
                >
                  <a
                    className="small-card__link"
                    href="/card/8"
                    onClick={[Function]}
                  >
                    Heart of Gold
                  </a>
                </h3>
                <p
                  className="small-card__price"
                >
                  100000000
                  $
                </p>
              </div>
              <button
                className="small-card__button"
                onClick={[Function]}
                type="button"
              >
                Add
                 
              </button>
            </article>
          </div>
          <div
            className="catalog__more"
          >
            <button
              className="catalog__button"
              onClick={[Function]}
              type="button"
            >
              Show more
            </button>
          </div>
        </section>
      </div>
    </div>
  </main>,
  <footer
    className="page-footer"
  >
    <div
      className="page-footer__wrapper"
    >
      <a
        className="page-footer__logo"
        href="/"
        onClick={[Function]}
      >
        Starshipsshop
      </a>
      <div
        className="page-footer__copyright"
      >
        <p
          className="page-footer__text"
        >
          © 2020 Stars ships shop Ltd.
        </p>
      </div>
    </div>
  </footer>,
]
`;
